src/puya/awst/serialize.py

```diff
@@ -86,4 +88,7 @@ def awst_to_json(awst: nodes.AWST) -> str:
 
 
 def awst_from_json(json: str) -> nodes.AWST:
-    return _get_converter().loads(json, nodes.AWST)  # type: ignore[type-abstract]
+    try:
+        return _get_converter().loads(json, nodes.AWST)  # type: ignore[type-abstract]
+    except (ClassValidationError, IterableValidationError, BaseException) as err:
```

Comment:
Never catch `BaseException` - it'll catch things like [KeyboardInterrupt](https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt)

In general it's not even recommended to catch `Exception`, but sometimes that's warranted, and this might be one of those cases.

Reply:
Removed BaseException from clauses